app/src/main/java/cm/aptoide/pt/billing/authorization/AuthorizationFactory.java

```diff
@@ -1,75 +1,47 @@
-        return new Authorization(paymentMethodId, payerId, status);
-    }
-  }
-
-  public List<Authorization> map(PaymentAuthorizationsResponse response, String payerId,
-      int paymentId) {
+  public static final String WEB = "WEB";
+  public static final String PAYPAL_SDK = "PAYPAL_SDK";
+  public static final String ADYEN_SDK = "ADYEN_SDK";
 
-    final List<Authorization> authorizations = new ArrayList<>();
-    if (response != null
-        && response.isOk()
-        && response.getAuthorizations() != null
-        && !response.getAuthorizations()
-        .isEmpty()) {
+  public Authorization create(String id, String customerId, String type,
+      Authorization.Status status, String url, String redirectUrl, String metadata, Price price,
+      String description, String transactionId, String session) {
 
-      for (PaymentAuthorizationResponse authorizationResponse : response.getAuthorizations()) {
-        authorizations.add(create(authorizationResponse.getPaymentId(),
-            Authorization.Status.valueOf(authorizationResponse.getAuthorizationStatus()), payerId,
-            authorizationResponse.getUrl(), authorizationResponse.getSuccessUrl()));
-      }
-    } else {
-      authorizations.add(create(paymentId, getStatus(response), payerId, "", ""));
+    if (type == null) {
+      return new Authorization(id, customerId, status, transactionId);
     }
 
-    return authorizations;
+    switch (type) {
+      case WEB:
+        return new WebAuthorization(id, customerId, status, url, redirectUrl, transactionId);
+      case PAYPAL_SDK:
+        return new PayPalAuthorization(id, customerId, status, transactionId, metadata, price,
+            description);
+      case ADYEN_SDK:
+        return new AdyenAuthorization(id, customerId, status, transactionId, session, metadata);
+      default:
+        return new Authorization(id, customerId, status, transactionId);
+    }
   }
 
-  public Authorization map(PaymentAuthorizationResponse response, String payerId, int paymentId) {
-
-    if (response != null && response.isOk()) {
-
-      return create(response.getPaymentId(),
-          Authorization.Status.valueOf(response.getAuthorizationStatus()), payerId,
-          response.getUrl(), response.getSuccessUrl());
-    } else {
-      return create(paymentId, Authorization.Status.UNKNOWN_ERROR, payerId, "", "");
+  public String getType(Authorization authorization) {
+    if (authorization instanceof AdyenAuthorization) {
+      return ADYEN_SDK;
     }
-  }
 
-  private Authorization.Status getStatus(PaymentAuthorizationsResponse response) {
+    if (authorization instanceof PayPalAuthorization) {
+      return PAYPAL_SDK;
+    }
 
-    if (response.getAuthorizations() != null && response.getAuthorizations()
-        .isEmpty()) {
-      return Authorization.Status.INACTIVE;
+    if (authorization instanceof WebAuthorization) {
+      return WEB;
     }
 
-    return Authorization.Status.UNKNOWN_ERROR;
+    return null;
```

Comment:
Question: Shouldn't this return a something else than null? Even though null type is being handled in the create method.

Reply:
Won't fix.